### PR TITLE
Language modeling flex heads

### LINKED ISSUE #53: Language modeling head for flexible head classes
Hi! I have many language adapters, each trained with masked language modeling on different (English) datasets. 

I want to be able to load *1* BERT model, load each of the adapters, and then decide which adapter to use on any given forward pass. This would save on memory and loading time, as opposed to loading a separate BERT for each adapter. This *seems* possible -- here's what I'm doing:

```
model_name = 'bert-base-cased'
model = BertForMaskedLM.from_pretrained(model_name)
model.load_adapter(ADAPTER1)
model.load_adapter(ADAPTER2)
# In practice, I have many more adapters

output1 = model(input_ids, token_type_ids, attention_mask, adapter_names=[ADAPTER1])
output2 = model(input_ids, token_type_ids, attention_mask, adapter_names=[ADAPTER2])
```

However, I am getting different outputs than having one model per adapter. For example, **`output1` above is different from `output1` below, and `output2` above is different from `output2` below.**
```
model_name = 'bert-base-cased'

model1 = BertForMaskedLM.from_pretrained(model_name)
model1.load_adapter(ADAPTER1)
output1 = model1(input_ids, token_type_ids, attention_mask, adapter_names=[ADAPTER1])

model2 = BertForMaskedLM.from_pretrained(model_name)
model2.load_adapter(ADAPTER2)
output2 = model2(input_ids, token_type_ids, attention_mask, adapter_names=[ADAPTER2])
```

I'm trying to read https://github.com/Adapter-Hub/adapter-transformers/blob/master/src/transformers/adapter_model_mixin.py#L339 to see why this would be the case. Is this expected behavior? Am I doing something wrong?

Thanks!

### PR DESCRIPTION
**Pull request by calpt**
Closes #53.

Waiting for #208.

---

This PR adds three different prediction heads for `XModelWithHeads` classes, depending on the model architecture:

- `add_causal_lm_head()` adds a causal LM head for classes that support this type of head in transformers, e.g. GPT-2, BERT, ...
- `add_masked_lm_head()` adds a masked LM head for models with MLM, e.g. BERT, RoBERTa, ...
- `add_seq2seq_lm_head()` adds a sequence-to-sequence LM head for encoder-decoder models, e.g. BART

All heads can be automatically converted from their respective static-head counterparts (e.g. `seq2seqlm` from `BartForConditionalGeneration`).

To ensure that all conversions work as expected, a new test module was added in `test_adapter_conversion.py`.

### CODE REVIEW

**Review by hSterz — APPROVED** [review]
Looks good
